src/scene/materials/standard-node-material.js

```diff
@@ -0,0 +1,189 @@
+            }
+        }
+    },
+
+    /**
+     * @private
+     * @function
+     * @name pc.StandardNodeMaterial#overrideGraphParameter
+     * @description Sets a shader graph parameter on a StandardNodeMaterial.
+     * @param {string} name - The name of the parameter to set.
+     * @param {number|number[]|pc.Texture} data - The value for the specified parameter.
+     */
+    overrideGraphParameter: function (name, data) {
+        if (this.shaderGraphChunk) {
+            var rootShaderGraph = this.shaderGraphChunk;
+            var portName = rootShaderGraph.getIoPortUniformName(name);
+            if (portName) {
+                this._graphParamOverrides[name] = { scopeId: null, data: data };
+                // StandardMaterial.prototype.setParameter.call(this, portName, data);
+            } else {
+                console.warn('not a shaderGraphChunk parameter - did you want to call setParameter()?');
+            }
+        }
+    },
+
+    updateUniforms: function () {
+        StandardMaterial.prototype.updateUniforms.call(this);
+
+        if (this.shaderGraphChunk) {
+            this.shaderGraphChunk.updateUniforms();
+        }
+    },
+
+    updateShader: function (device, scene, objDefs, staticLightList, pass, sortedLights) {
+        var prefilteredCubeMap128 = StandardMaterial.prototype._updateShaderGlobals.call(this, device, scene);
+
+        // Minimal options for Depth and Shadow passes
+        var minimalOptions = pass > SHADER_FORWARDHDR && pass <= SHADER_PICK;
+        var options = minimalOptions ? standard.optionsContextMin : standard.optionsContext;
+
+        if (minimalOptions)
+            this.shaderOptBuilder.updateMinRef(options, device, scene, this, objDefs, staticLightList, pass, sortedLights, prefilteredCubeMap128);
+        else
+            this.shaderOptBuilder.updateRef(options, device, scene, this, objDefs, staticLightList, pass, sortedLights, prefilteredCubeMap128);
+
+        // add shader graph chunk to options
+        if (this.shaderGraphChunk) {
+            options._shaderGraphChunk = this.shaderGraphChunk;
+
+            options._graphSwitchOverrides = this._graphSwitchOverrides;
+            options._graphParamOverrides = this._graphParamOverrides;
+
+            // shader graph chunk texture sampling can switch to this
+            options.useRgbm = true;
+        }
+
+        if (this.onUpdateShader) {
+            options = this.onUpdateShader(options);
+        }
+
+        var library = device.getProgramLibrary();
+        this.shader = library.getProgram('standardnode', options);
+
+        if (!objDefs) {
+            this.clearVariants();
+            this.variants[0] = this.shader;
+        }
+
+        this.dirtyShader = false;
+    }
+});
+
+Object.defineProperty(StandardNodeMaterial.prototype, 'shaderGraphChunk', {
+    get: function () {
+        return this._shaderGraphChunk;
+    },
+    set: function (value) {
+        if (!value || this._shaderGraphChunk === value)
```

Comment:
just curious about the !value here. Should the code that removes current chunk execute? When I set something to null, I'd expect current value to be removed.

Reply:
yeah not sure why its here - I shall remove it

Reply:
removed